src/vellum/workflows/events/types.py

```diff
@@ -73,20 +74,30 @@ class PromptDeploymentParentContext(BaseDeploymentParentContext):
 
 class NodeParentContext(BaseParentContext):
     type: Literal["WORKFLOW_NODE"] = "WORKFLOW_NODE"
-    node_definition: Type['BaseNode']
+    node_definition: Union[Type["BaseNode"], CodeResourceDefinition]
 
     @field_serializer("node_definition")
-    def serialize_node_definition(self, definition: Type, _info: Any) -> Dict[str, Any]:
-        return serialize_type_encoder(definition)
+    def serialize_node_definition(
+        self, definition: Union[Type["BaseNode"], CodeResourceDefinition], _info: Any
+    ) -> Dict[str, Any]:
+        if isinstance(definition, type):
+            return serialize_type_encoder(definition)
+        else:
+            return definition.model_dump()
 
 
 class WorkflowParentContext(BaseParentContext):
     type: Literal["WORKFLOW"] = "WORKFLOW"
-    workflow_definition: Type['BaseWorkflow']
+    workflow_definition: Union[Type["BaseWorkflow"], CodeResourceDefinition]
 
     @field_serializer("workflow_definition")
-    def serialize_workflow_definition(self, definition: Type, _info: Any) -> Dict[str, Any]:
-        return serialize_type_encoder(definition)
+    def serialize_workflow_definition(
```

Comment:
should allow for both __class__ and the definition that is passed through to be handled correctly